content/WebAuthn/Concepts/Using_WebAuthn_for_Signing.adoc

```diff
@@ -0,0 +1,25 @@
+= Signing with WebAuthn
+
+While WebAuthn was designed primarily for authentication, the framework can be extended to support signing with keypairs generated on authenticators without requiring custom clients for end users.
+
+== Overview
+During a standard WebAuthn interaction, a WebAuthn authenticator creates a unique asymmetric keypair with a relying party at registration, with the private key residing only on the authenticator and the public key sent back. As part of the registration event, a challenge from the relying party is signed with the unique keypair and returned. Afterwards, an authentication event uses this key pair to sign a challenge which includes a random string provided by the relying party. In short, the registration and authenticaiton events consists of a cryptographic signing process with the public key returned, which is then used to sign a data set which is provided by the relying party. Replacing the randomly generated challenge with a hash of a file to be signed allows this signature flow to be applied to actual documentation, with the public key also available for verification after the fact.
+
+The main advantages to this flow is it allows for any cryptographic algorithm supported by the WebAuthn framework to be used, and can leverage the existing WebAuthn support in browsers or local APIs without the need to apply code changes. All of the modifications required would be limited to the WebAuthn libraries/servers owned by the relying party, permitting a service to implement arbitrary file signing without requiring significant user commitment.
+
+== High-Level Architecture
+This section will discuss at a high level the architecture necessary for adapting a WebAuthn flow for file signing. The actual implementation will depend on the environment, WebAuthn server or library and the WebAuthn client (brower or local) in use, and is outside of the scope of this document.
+
+The method for leveraging WebAuthn for signature operations on arbitrary files is built around the FIDO Signature Creation flow, comprised of the link:https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred[MakeCredential] and link:https://www.w3.org/TR/webauthn-2/#sctn-op-get-assertion[GetAssertion] operations.
```

Comment:
Just thought this can be said a bit simpler and more concisely. :slightly_smiling_face:
```suggestion
The method for leveraging WebAuthn to sign arbitrary files is built around the FIDO Signature Creation flow, comprised of the link:https://www.w3.org/TR/webauthn-2/#sctn-op-make-cred[MakeCredential] and link:https://www.w3.org/TR/webauthn-2/#sctn-op-get-assertion[GetAssertion] operations.
```